### PR TITLE
Bundle ServiceMonitor CRD with the chart

### PR DESCRIPTION
**Pull request by diversario**
This allows installing Cilium with service monitor enabled
even if Prometheus hasn't been installed yet.

By including the ServiceMonitor CRD under `crds/` directory we ensure
that Helm will install it first iff the cluster has no existing
ServiceMonitor CRD. This behavior is documented at
https://helm.sh/docs/topics/charts/#custom-resource-definitions-crds

Fixed errors such as

```sh
% helm install cilium ./cilium --namespace kube-system --set global.prometheus.enabled=true--set --set global.prometheus.serviceMonitor.enabled=true
Error: unable to build kubernetes objects from release manifest: unable to recognize "": no matches for kind "ServiceMonitor" in version "monitoring.coreos.com/v1"
```

This requires Helm 3; Helm 2 will ignore the crds/ directory. The CRD is copied from [this commit of the helm/charts repo](https://github.com/helm/charts/blob/6c7638eef1730a6e4de48f37fa3262a427106885/stable/prometheus-operator/crds/crd-servicemonitor.yaml).

Signed-off-by: Ilya Shaisultanov <ilya.shaisultanov@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

Fixes: #issue-number

```release-note
Allow enabling ServiceMonitor without Prometheus installed.
```

### CODE REVIEW

**Review by michi-covalent — APPROVED** [review]
thank you for taking care of this

File: install/kubernetes/cilium/crds/crd-servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,460 @@
+# https://raw.githubusercontent.com/coreos/prometheus-operator/release-0.38/example/prometheus-operator-crd/monitoring.coreos.com_servicemonitors.yaml
```
Comment (aanm) [inline install/kubernetes/cilium/crds/crd-servicemonitor.yaml]: This should be under a conditional branch. This should only be installed if the serviceMonitor is enabled.

**Review by aanm — CHANGES_REQUESTED** [review]

File: install/kubernetes/cilium/crds/crd-servicemonitor.yaml (hunk 1 above)
Comment (diversario) [reply install/kubernetes/cilium/crds/crd-servicemonitor.yaml]: That is a good point, but possibly not achievable with Helm:

> CRD files cannot be templated. They must be plain YAML documents.

I shall see if there's any way that could make this conditional.

Comment (aanm) [reply install/kubernetes/cilium/crds/crd-servicemonitor.yaml]: :exploding_head:

**Comment by aanm** [comment]
Discussed offline how to fix this. Waiting for implementation.

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Pull request merged**